packages/node/src/integrations/tracing/fastify.ts

```diff
@@ -61,14 +61,27 @@ const _fastifyIntegration = (() => {
  */
 export const fastifyIntegration = defineIntegration(_fastifyIntegration);
 
+interface FastifyHandlerOptions {
+  /**
+   * Callback method deciding whether error should be captured and sent to Sentry
+   * @param error Captured middleware error
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  shouldHandleError?(this: void, request: any, reply: any, error: Error): boolean;
```

Comment:
It is better to use explicit types instead of any
import type { FastifyRequest, FastifyReply } from 'fastify';

Reply:
for that we need to depend on the `fastify` package, which we cannot do in the generic node package.

Reply:
// grumbling-mod-on
When someone was too lazy to take the framework packages out of the general node package :) Although this was proposed for version 8
// grumbling-mod-off

Perhaps, a dummy interface can be used. After all, only a little piece of it is used.
Or the generic type.

Reply:
> When someone was too lazy to take the framework packages out of the general node package :) Although this was proposed for version 8

It was a conscious decision not to do this.

> Perhaps, a dummy interface can be used. After all, only a little piece of it is used.

Which is exactly as fragile as `any`, if not more, since it gives a false sense of security and disables all of the ts-eslint checking.

Reply:
I have to disagree with you. The `any` can completely disable checks, but the interface will force you as a developer to add/edit the necessary entry. This means that some commands will not be called accidentally. As below, when the `statusCode` may be undefined.

Anyway, this is just my opinion. However, the number of open issues with the new version 8 makes it clear that it is possible. In any case, it is your conscious decision.

Reply:
I think you're making wrong assumptions about how our linting works. When we define stuff as `any`, yes typescript will not check it anymore, however, eslint will still not let us access any fields without narrowing down and checking for their exact type first.

I encourage you to pull the repo and try it out.